src/lib/base/buf_comp.h

```diff
@@ -59,57 +59,57 @@ class BOTAN_DLL Buffered_Computation
          {
          for(size_t i = 0; i != sizeof(T); ++i)
             {
-            byte b = get_byte(i, in);
+            uint8_t b = get_byte(i, in);
             add_data(&b, 1);
             }
          }
 
       /**
       * Add new input to process.
       * @param str the input to process as a std::string. Will be interpreted
-      * as a byte array based on
+      * as a uint8_t array based on
       * the strings encoding.
       */
       void update(const std::string& str)
          {
-         add_data(reinterpret_cast<const byte*>(str.data()), str.size());
+         add_data(reinterpret_cast<const uint8_t*>(str.data()), str.size());
          }
 
       /**
       * Process a single byte.
-      * @param in the byte to process
+      * @param in the uint8_t to process
       */
-      void update(byte in) { add_data(&in, 1); }
+      void update(uint8_t in) { add_data(&in, 1); }
 
       /**
       * Complete the computation and retrieve the
       * final result.
-      * @param out The byte array to be filled with the result.
+      * @param out The uint8_t array to be filled with the result.
```

Comment:
Yeah, some of these I had reverted already prior to my commit, but I left many where either is technically valid. But "byte string" is easier to parse than "uint8_t string", and that it is a uint8_t vs a future byte type vs unsigned char is kind of an implementation detail. I will revisit the changes at least to the comments for the major APIs.